autogpt_platform/backend/schema.prisma

```diff
@@ -136,14 +239,14 @@ model AgentGraphExecution {
   executionStatus AgentExecutionStatus @default(COMPLETED)
 
   agentGraphId      String
-  agentGraphVersion Int        @default(1)
-  AgentGraph        AgentGraph @relation(fields: [agentGraphId, agentGraphVersion], references: [id, version], onDelete: Cascade)
+  agentGraphVersion Int    @default(1)
+  AgentGraph        Agent  @relation(fields: [agentGraphId, agentGraphVersion], references: [id, version], onDelete: Cascade)
 
   AgentNodeExecutions AgentNodeExecution[]
 
-  // Link to User model
-  userId String
-  user   User   @relation(fields: [userId], references: [id], onDelete: Cascade)
+  // This is so we can track which user executed the agent graph for accounting purposes.
+  executedByUserId String
+  executedByUser   User   @relation(fields: [executedByUserId], references: [id], onDelete: Cascade)
```

Comment:
As far as I know, this isn't just for accounting purposes. Also essential for access control.

Reply:
Yes absolutely, great idea. I'll update the comment too